File: app/src/main/java/me/echeung/moemoekyun/viewmodel/SongListViewModel.java
package me.echeung.moemoekyun.viewmodel;

import androidx.databinding.BaseObservable;
import androidx.databinding.Bindable;
import androidx.databinding.library.baseAdapters.BR;

public class SongListViewModel extends BaseObservable {

    private String query;
    private boolean hasResults;

    @Bindable
    public String getQuery() {
        return query;
    }

    public void setQuery(String query) {
        this.query = query;
        notifyPropertyChanged(BR.query);
    }

    public void clearQuery() {
        setQuery(null);
    }

    @Bindable
    public boolean getHasResults() {
        return hasResults;
    }

    public void setHasResults(boolean hasResults) {
        this.hasResults = hasResults;
        notifyPropertyChanged(BR.hasResults);
    }

}

File: app/src/main/java/me/echeung/moemoekyun/viewmodel/SongListViewModel.kt
package me.echeung.moemoekyun.viewmodel

import androidx.databinding.BaseObservable
import androidx.databinding.Bindable
import androidx.databinding.library.baseAdapters.BR

class SongListViewModel : BaseObservable() {

    @get:Bindable
    var query: String? = null
        set(query) {
            field = query
            notifyPropertyChanged(BR.query)
        }

    @get:Bindable
    var hasResults: Boolean = false
        set(hasResults) {
            field = hasResults
            notifyPropertyChanged(BR.hasResults)
        }

    fun clearQuery() {
        query = null
    }
}
